example/simple_compute.cpp

```diff
@@ -0,0 +1,302 @@
+    vkb::InstanceDispatchTable inst_disp;
+    vkb::Device device;
+    vkb::DispatchTable disp;
+};
+
+struct Data {
+    VkQueue queue;
+
+    VkPipelineLayout pipeline_layout;
+    VkPipeline compute_pipeline;
+
+    VkCommandPool command_pool;
+    VkCommandBuffer command_buffer;
+
+    VkBuffer buffer_a;
+    VkDeviceMemory memory_a;
+    VkBuffer buffer_b;
+    VkDeviceMemory memory_b;
+
+    VkDescriptorSet descriptor_set;
+    VkDescriptorPool descriptor_pool;
+    VkDescriptorSetLayout descriptor_set_layout;
+};
+
+int device_initialization(Init& init) {
+    vkb::InstanceBuilder instance_builder;
+    auto instance_ret = instance_builder.use_default_debug_messenger().request_validation_layers().build();
+    if (!instance_ret) {
+        std::cout << instance_ret.error().message() << "\n";
+        return -1;
+    }
+    init.instance = instance_ret.value();
+
+    init.inst_disp = init.instance.make_table();
+
+    vkb::PhysicalDeviceSelector phys_device_selector(init.instance);
+    // Skip vk-bootstrap trying to create WSI for you
+    auto phys_device_ret = phys_device_selector.defer_surface_initialization().select();
+    if (!phys_device_ret) {
+        std::cout << phys_device_ret.error().message() << "\n";
+        return -1;
+    }
+    vkb::PhysicalDevice physical_device = phys_device_ret.value();
+
+    vkb::DeviceBuilder device_builder{ physical_device };
+    auto device_ret = device_builder.build();
+    if (!device_ret) {
+        std::cout << device_ret.error().message() << "\n";
+        return -1;
+    }
+    init.device = device_ret.value();
+
+    init.disp = init.device.make_table();
+
+    return 0;
+}
+
+int get_queues(Init& init, Data& data) {
+    // Compute can be done on a Grahpics queue
+    // See require_dedicated_compute_queue() if want a dedicated compute queue
+    auto gq = init.device.get_queue(vkb::QueueType::graphics);
+    if (!gq.has_value()) {
+        std::cout << "failed to get queue: " << gq.error().message() << "\n";
+        return -1;
+    }
+    data.queue = gq.value();
+    return 0;
+}
+
+std::vector<char> readFile(const std::string& filename) {
+    std::ifstream file(filename, std::ios::ate | std::ios::binary);
+
+    if (!file.is_open()) {
+        throw std::runtime_error("failed to open file!");
+    }
+
+    size_t file_size = (size_t)file.tellg();
+    std::vector<char> buffer(file_size);
+
+    file.seekg(0);
+    file.read(buffer.data(), static_cast<std::streamsize>(file_size));
+    file.close();
+
+    return buffer;
+}
+
+
+int create_descriptor(Init& init, Data& data) {
+    VkDescriptorPoolSize pool_size = { VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, 2 };
+
+    VkDescriptorPoolCreateInfo pool_info = {};
+    pool_info.sType = VK_STRUCTURE_TYPE_DESCRIPTOR_POOL_CREATE_INFO;
+    pool_info.flags = 0;
+    pool_info.maxSets = 1;
+    pool_info.poolSizeCount = 1;
+    pool_info.pPoolSizes = &pool_size;
+    init.disp.createDescriptorPool(&pool_info, nullptr, &data.descriptor_pool);
```

Comment:
Not even a simple return -1 if not success?

The lack of error checking makes the int return value feel pointless. Could do away with all error checking but like thats a basic rule of thumb - check the return value.

Reply:
I guess the example is so simple, this should not fail, not the goal for people to use these bare old ways to create a pool (hope the delete this code for something better)

I guess also wanted to squeeze it in 300 lines lol

Reply:
At that point might as well just remove the error checks anywhere that isnt vkbootstrap. Which honestly is fine, only place worth checking is vkqueuesubmit since everything else should just work.